processing/src/main/java/io/druid/query/CacheStrategy.java

```diff
@@ -27,12 +27,11 @@
 public interface CacheStrategy<T, CacheType, QueryType extends Query<T>>
 {
   /**
-   * Computes the cache key for the given query
+   * Computes the cache key for the associated query
    *
-   * @param query the query to compute a cache key for
    * @return the cache key
    */
-  byte[] computeCacheKey(QueryType query);
+  byte[] computeCacheKey();
```

Comment:
This will impact anyone that has a custom query type as an extension. However, I don't believe anyone does. In any case, I think we should bring this up in the user groups to see if these changes will impact anyone in the community.